src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/openapi/TokenCertificatesApiController.java

```diff
@@ -96,4 +96,24 @@ public TokenCertificatesApiController(TokenCertificateService tokenCertificateSe
         CertificateDetails certificateDetails = certificateDetailsConverter.convert(certificateInfo);
         return new ResponseEntity<>(certificateDetails, HttpStatus.OK);
     }
+
+    @Override
+    @PreAuthorize("hasAnyAuthority('IMPORT_AUTH_CERT', 'IMPORT_SIGN_CERT')")
+    public ResponseEntity<CertificateDetails> importExistingCertificate(String hash) {
```

Comment:
Maybe `importCertificateFromToken` would be even more accurate? Since core just searches through all existing tokens for the cert (could also be a software token right?)

Reply:
Yeah, maybe so. I honestly don't even know if that is possible or not (import from software token)